crates/ott-common/src/discovery/dns.rs

```diff
@@ -21,6 +25,44 @@ pub struct DnsDiscoveryConfig {
     pub polling_interval: Option<Duration>,
 }
 
+trait FromValue {
+    fn from_value(value: serde_json::Value) -> Result<DnsDiscoveryConfig, serde_json::Error>;
+}
+
+impl FromValue for DnsDiscoveryConfig {
+    fn from_value(mut value: serde_json::Value) -> Result<DnsDiscoveryConfig, serde_json::Error> {
+        if value.get("dns_server").is_none() {
+            let mut kv = serde_json::Map::new();
+            kv.insert(
+                "dns_server".to_string(),
+                serde_json::Value::from_str("").unwrap(),
+            );
+            let _value = value.as_object_mut().insert(&mut kv);
+        }
+
+        serde_json::from_value(value)
+    }
+}
```

Comment:
Modifying the config is a bad idea. It's supposed to be immutable. Also, this doesn't actually change the way `DnsDiscoveryConfig` is deserialized.

You want [`#[serde(default)]`](https://serde.rs/attr-default.html) (see `polling_interval`)